.github/workflows/gen-check.yaml

```diff
@@ -0,0 +1,17 @@
+name: gen-check
+
+on: [push, pull_request]
+
+jobs:
+  gen-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Install yq
+      run: |
+        go install github.com/mikefarah/yq/v4@v4.45.1
```

Comment:
nit: you shouldn't need to explicitly install yq since it's included in GH actions tooling: https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md#tools but doesn't matter much if you want to keep this.

Reply:
oh, that's a great catch! will remove that step, it takes forever

Reply:
yay, job execution time went from 41s->3s :partying_face: